### ISSUE TITLE
Lint/UnescapedBracketInRegexp false positive

### ISSUE THREAD

**Issue by camero2734** [rubocop/rubocop#13511]
We have a simple regex that uses lookbehind, and this cop is incorrectly flagging it as having an unescaped bracket

The regex is:
```rb
/(?<=[<>=:])/
```

and it wants to autocorrect it to:
```rb
/(?<=[<>=:\])/
```
which is a broken regex

--------

## Expected behavior
No error to occur

## Actual behavior

```
7836422be53c:/project# bundle exec rubocop --debug test.rb
For /project: configuration from /project/.rubocop.yml
configuration from /usr/local/bundle/gems/rubocop-faker-1.2.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-faker-1.2.0/config/default.yml
Default configuration from /usr/local/bundle/gems/rubocop-1.68.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-performance-1.19.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-performance-1.19.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.9.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.9.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rspec-3.2.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rspec-3.2.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-factory_bot-2.26.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-factory_bot-2.26.1/lib/../config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /project/test.rb
Loading cache from /root/.cache/rubocop_cache/403fc3668b52ef4320081cf59924af134c591ad0/e52892fe07ef1f949cb5c4bc40fe9af2414219f9/ed324501581847e0ae909c06f5d6bd2c0913663e
I

Offenses:

test.rb:1:20: I: [Correctable] Lint/UnescapedBracketInRegexp: Regular expression has ] without escape.
regexp = /(?<=[<>=:])/
                   ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.1824979051016271 seconds
```

## Steps to reproduce the problem
Test file:
```rb
regexp = /(?<=[<>=:])/
puts regexp
```

Changing the order seems to "fix" it, for example this doesn't have a violation:
```rb
regexp = /(?<=[=:<>])/
puts regexp
```

## RuboCop version

```
7836422be53c:/project# bundle exec rubocop -V
1.68.0 (using Parser 3.3.6.0, rubocop-ast 1.33.0, analyzing as Ruby 3.2, running on ruby 3.2.6) [aarch64-linux-musl]
  - rubocop-factory_bot 2.26.1
  - rubocop-faker 1.2.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.9.1
  - rubocop-rspec 3.2.0
```

**Comment by dvandersluis** [comment]
There is a bug in regexp_parser which is parsing `/(?<=...>)/` as a named group instead of a lookbehind. I opened https://github.com/ammar/regexp_parser/issues/93 to track it.
